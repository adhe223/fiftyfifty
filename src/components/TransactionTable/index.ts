export { default } from './TransactionTable';
export * from './TransactionTable';
