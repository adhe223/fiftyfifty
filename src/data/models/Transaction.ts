import Person from './Person';

export default interface Transaction {
  id: number;
  description: string;
  date: Date;
  amount: number;
  person: Person;
  settled: Boolean;
};
