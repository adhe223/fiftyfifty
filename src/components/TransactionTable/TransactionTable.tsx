import * as React from 'react';

import Transaction from '../../data/models/Transaction';

export interface TransactionTableProps {
  transactions: Transaction[];
}

const TransactionTable = (props: TransactionTableProps) => {
  const rows = props.transactions.map((transaction: Transaction, index) => (
    <tr key={'row-' + index} className="transaction-table--row">
      <td key={'row-' + index + 'td-1'}>{transaction.date.toString()}</td>
      <td key={'row-' + index + 'td-2'}>{transaction.person.name}</td>
      <td key={'row-' + index + 'td-3'}>{transaction.description}</td>
      <td key={'row-' + index + 'td-4'}>{transaction.amount}</td>
    </tr>
  ));

  return (
    <table className="transaction-table">
      <tbody>
        <tr className="transaction-table--header-row" key="tr-header-row">
          <th>Date</th>
          <th>Person</th>
          <th>Description</th>
          <th>Amount</th>
        </tr>
        {rows}
      </tbody>
    </table>
  );
};

export default TransactionTable;
