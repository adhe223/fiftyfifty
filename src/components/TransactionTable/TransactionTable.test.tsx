import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import TransactionTable from './TransactionTable';
import Transaction from '../../data/models/Transaction';
import { TransactionTableProps } from './TransactionTable';

describe('TransactionTable', () => {
  describe('render', () => {
    let wrapper: ShallowWrapper<TransactionTableProps, {}>;

    beforeEach(() => {
      const alex = {
        id: 'alexId',
        name: 'Alex'
      };
      const cindy = {
        id: 'cindyId',
        name: 'Cindy'
      };

      const transactions: Transaction[] = [
        {
          id: 1,
          description: 'Test 1',
          amount: 1,
          date: new Date('1/1/2001'),
          personId: alex.id,
          settled: false
        },
        {
          id: 2,
          description: 'Test 2',
          amount: 2,
          date: new Date('2/2/2002'),
          personId: cindy.id,
          settled: false
        }
      ];

      wrapper = shallow(<TransactionTable transactions={transactions} />);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
