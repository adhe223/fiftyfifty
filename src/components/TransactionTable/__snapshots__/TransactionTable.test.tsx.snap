// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionTable render should render correctly 1`] = `
<table
  className="transaction-table"
>
  <tbody>
    <tr
      className="transaction-table--header-row"
      key="tr-header-row"
    >
      <th>
        Date
      </th>
      <th>
        Person
      </th>
      <th>
        Description
      </th>
      <th>
        Amount
      </th>
    </tr>
    <tr
      className="transaction-table--row"
      key="row-0"
    >
      <td
        key="row-0td-1"
      >
        Mon Jan 01 2001 00:00:00 GMT-0800 (PST)
      </td>
      <td
        key="row-0td-2"
      >
        Alex
      </td>
      <td
        key="row-0td-3"
      >
        Test 1
      </td>
      <td
        key="row-0td-4"
      >
        1
      </td>
    </tr>
    <tr
      className="transaction-table--row"
      key="row-1"
    >
      <td
        key="row-1td-1"
      >
        Sat Feb 02 2002 00:00:00 GMT-0800 (PST)
      </td>
      <td
        key="row-1td-2"
      >
        Cindy
      </td>
      <td
        key="row-1td-3"
      >
        Test 2
      </td>
      <td
        key="row-1td-4"
      >
        2
      </td>
    </tr>
  </tbody>
</table>
`;
